### DISCUSSION TITLE
Parameter / Flag conventions

### DISCUSSION THREAD

**Discussion by francescobianco** [francescobianco/shell-output-guidelines#12] [General]
Which concerns to  parameter / flag conventions

**Comment by Potherca** [comment]
The [Command Line Interface Guidelines](https://clig.dev/) section regarding [arguments and flags](https://clig.dev/#arguments-and-flags) has some mentions:

> Use standard names for flags, if there is a standard. If another commonly used command uses a flag name, it’s best to follow that existing pattern. That way, a user doesn’t have to remember two different options (and which command it applies to), and users can even guess an option without having to look at the help text.
> 
> Here’s a list of commonly used options:
> 
> -   `-a`, `--all`: All. For example, `ps`, `fetchmail`.
> -   `-d`, `--debug`: Show debugging output.
> -   `-f`, `--force`: Force. For example, `rm -f` will force the removal of files, even if it thinks it does not have permission to do it. This is also useful for commands which are doing something destructive that usually require user confirmation, but you want to force it to do that destructive action in a script.
> -   `--json`: Display JSON [output](https://clig.dev/#output). See the output section.
> -   `-h`, `--help`: Help. This should only mean [help](https://clig.dev/#help). See the help section.
> -   `--no-input`: See the [interactivity](https://clig.dev/#interactivity) section.
> -   `-o`, `--output`: Output file. For example, `sort`, `gcc`.
> -   `-p`, `--port`: Port. For example, `psq`l, `ssh`.
> -   `-q`, `--quiet`: Quiet. Display less output. This is particularly useful when displaying output for humans that you might want to hide when running in a script.
> -   `-u`, `--user`: User. For example, `ps`, `ssh`.
> -   `--version`: Version.
> -   `-v`: This can often mean either verbose or version. You might want to use `-d` for verbose and this for version, or for nothing to avoid confusion.

Although I'm not sure I agree with all of these, they provide a good starting point.

**Comment by Potherca** [comment]
It might be worth looking at the most used CLI tools and create some sort of overview of who uses which for what.

https://popcon.debian.org/ would be a good starting place regarding popularity / usage.

Some other common ones

- `-c`, `--config` to load a configuration file
- `-r`, `--recursive` operate recursively on directories

Something else I have noticed is that, where older tools seem to prefer single character flags, newer tools prefer long parameter flags, sometimes to the point where they don't even have _any_ single character flags.
